lib/Dialect/Linalg/Transforms/LinalgTransforms.cpp

```diff
@@ -106,3 +111,38 @@ bool mlir::linalg::detail::isProducedByOpOfTypeImpl(
   }
   return false;
 }
+
+LogicalResult
+mlir::linalg::permuteGenericLinalgOp(PatternRewriter &rewriter, Operation *op,
+                                     ArrayRef<unsigned> permutation,
+                                     StringRef linalgMarker) {
+  // If permutation is empty, there is nothing to be done.
+  if (permutation.size() == 0)
+    return failure();
+
+  OpBuilder::InsertionGuard g(rewriter);
+  auto linOp = cast<LinalgOp>(op);
+  rewriter.setInsertionPoint(linOp);
+  ScopedContext scope(rewriter, linOp.getLoc());
+  auto permutationMap =
+      inversePermutation(AffineMap::getPermutationMap(permutation, scope.getContext()));
+  SmallVector<Attribute, 4> newIndexingMap;
+  auto indexingMaps = linOp.getAttrOfType<ArrayAttr>("indexing_maps").getValue();
+  for (unsigned i = 0, e = linOp.getNumInputsAndOutputs(); i != e; ++i) {
+    AffineMap m = indexingMaps[i].cast<AffineMapAttr>().getValue().compose(permutationMap);
+    newIndexingMap.push_back(AffineMapAttr::get(m));
+  }
+  auto itTypes = linOp.getAttrOfType<ArrayAttr>("iterator_types").getValue();
+  SmallVector<Attribute, 4> itTypesVector(itTypes.begin(), itTypes.end());
+  applyPermutationToVector(itTypesVector, permutation);
+  op->setAttr("indexing_maps",
+              rewriter.getArrayAttr(llvm::makeArrayRef(newIndexingMap)));
+  op->setAttr("iterator_types",
+              rewriter.getArrayAttr(llvm::makeArrayRef(itTypesVector)));
+  op->setAttr(LinalgTransforms::kLinalgTransformMarker,
+              rewriter.getStringAttr(linalgMarker));
```

Comment:
`rewriter. getStrArrayAttr` ?

Reply:
Done. It required slight rewrite of `itTypesVector` initialization.